src/ucp/rndv/proto_rndv.c

```diff
@@ -645,6 +645,7 @@ ucp_proto_rndv_bulk_init(const ucp_proto_multi_init_params_t *init_params,
     }
 
     rpriv->frag_mem_type = init_params->super.reg_mem_info.type;
+    rpriv->frag_sys_dev  = init_params->super.reg_mem_info.sys_dev;
```

Comment:
maybe use the struct ucp_memory_info_t in places we need to save mem_type+sys-dev, like here, and mpool_priv?

Reply:
I will address the comment in another PR since it will cause changes in several other places.